### PR TITLE
ES|QL tab added to Timeline

### PR DESCRIPTION
**Pull request by nastasha-solomon**
Fixes https://github.com/elastic/security-docs/issues/4025

Preview:
- [Use ES|QL to investigate events](https://security-docs_4065.docs-preview.app.elstc.co/guide/en/security/master/timelines-ui.html#esql-in-timeline) - New section
- [Additional ES|QL resources](https://security-docs_4065.docs-preview.app.elstc.co/guide/en/security/master/timelines-ui.html#esql-in-timeline-resources) - New section

### CODE REVIEW

**Review by michaelolo24 — APPROVED** [review]
These look great to me, thanks so much @nastasha-solomon !

**Comment by nastasha-solomon** [comment]
@michaelolo24 some additional ES|QL docs were created this week, so I added some additional ES|QL resources and restructured some of the existing docs about ES|QL in Timeline. Can you take a look at this draft next week when you're back? Thanks!

File: docs/events/timeline-ui-overview.asciidoc [hunk 1]
```diff
@@ -178,3 +178,29 @@ From the *Correlation* tab, you can also do the following:
...
+
+[discrete]
+[[esql-in-timeline]]
+== Use ES|QL to interact with alerts 
+
+preview::[]
+
+The {ref}/esql.html[Elasticsearch Query Language (ES|QL)] provides a powerful way to filter, transform, and analyze alert data stored in Elasticsearch. ES|QL queries use "pipes" to manipulate and transform data in a step-by-step fashion. This approach allows you to compose a series of operations, where the output of one operation becomes the input for the next, enabling complex data transformations and analysis.
+
+You can use ES|QL in Timeline by opening the **ES|QL** tab. From the tab, you can:
```
Comment (benironside) [inline docs/events/timeline-ui-overview.asciidoc:190]:
```suggestion
You can use ES|QL in Timeline by opening the **ES|QL** tab. From there, you can: 
```

File: docs/events/timeline-ui-overview.asciidoc [hunk 2]
```diff
@@ -178,3 +178,29 @@ From the *Correlation* tab, you can also do the following:
...
+[[esql-in-timeline]]
+== Use ES|QL to interact with alerts 
+
+preview::[]
+
+The {ref}/esql.html[Elasticsearch Query Language (ES|QL)] provides a powerful way to filter, transform, and analyze alert data stored in Elasticsearch. ES|QL queries use "pipes" to manipulate and transform data in a step-by-step fashion. This approach allows you to compose a series of operations, where the output of one operation becomes the input for the next, enabling complex data transformations and analysis.
+
+You can use ES|QL in Timeline by opening the **ES|QL** tab. From the tab, you can: 
+
+- Explore your alerts using the default query, or creating a custom one. The default query searches the Security alert index (`.alerts-security.alerts-default`) and  indices specified in the <<update-sec-indices,Security data view>> for alerts that were generated during the last 15 minutes. The query's output is limited to the top ten results.
```
Comment (benironside) [inline docs/events/timeline-ui-overview.asciidoc:192]:
```suggestion
- Explore your alerts using the default query, or a custom one. The default query searches the Security alert index (`.alerts-security.alerts-default`) and  indices specified in the <<update-sec-indices,Security data view>> for alerts that were generated during the last 15 minutes. The query's output is limited to the top ten results.
```

Comment (benironside) [reply docs/events/timeline-ui-overview.asciidoc:192]: The top ten alerts by what? Severity? Frequency?

File: docs/events/timeline-ui-overview.asciidoc [hunk 3]
```diff
@@ -178,3 +178,29 @@ From the *Correlation* tab, you can also do the following:
...
+- Click the help icon (image:images/esql-ref-button.png[Click the ES|QL reference button,20,20]) on the far right side of the query editor to open the in-product reference documentation for all ES|QL commands and functions.  
+- Visualize query results using {kibana-ref}/discover.html[Discover] functionality that's available in the tab.
```
Comment (benironside) [inline docs/events/timeline-ui-overview.asciidoc:194]:
```suggestion
- Visualize query results using {kibana-ref}/discover.html[Discover] functionality.
```

File: docs/events/timeline-ui-overview.asciidoc [hunk 4]
```diff
@@ -178,3 +178,29 @@ From the *Correlation* tab, you can also do the following:
...
+
+[role="screenshot"]
+image::images/esql-tab.png[a Timeline's ES|QL tab]
+
+[discrete]
+[[esql-in-timeline-resources]]
+=== Additional ES|QL resources 
+
+To get started using ES|QL, read the tutorial for {ref}/esql.html[using ES|QL in Kibana]. Much of the functionality that's offered in Kibana is also offered in Timeline.
```
Comment (benironside) [inline docs/events/timeline-ui-overview.asciidoc:203]:
```suggestion
To get started using ES|QL, read the tutorial for {ref}/esql.html[using ES|QL in Kibana]. Much of the functionality available in Kibana is also available in Timeline.  
```

**Review by benironside — COMMENTED** [review]
Looks great, very cool feature and clear description. Left a few suggestions for your consideration and one question. Thanks!

File: docs/events/timeline-ui-overview.asciidoc (hunk 2 above)
Comment (nastasha-solomon) [reply docs/events/timeline-ui-overview.asciidoc:192]: I'm not actually sure how the `[LIMIT](https://security-docs_4062.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/esql-commands.html#esql-limit)` command decides which rows/alerts count as the "top" ones. @michaelolo24 do you have any insight on this?

**Review by michaelolo24 — APPROVED** [review]
The added changes are informative and thank you for pointing out the reference button!

File: docs/events/timeline-ui-overview.asciidoc [hunk 5]
```diff
@@ -178,3 +178,29 @@ From the *Correlation* tab, you can also do the following:
 * Specify the date and time range that you want to investigate.
 * Reorder the columns and choose which fields to display.
 * Choose a data view and whether to show detection alerts only.
+
+[discrete]
+[[esql-in-timeline]]
+== Use {esql} to interact with alerts
```
Comment (nastasha-solomon) [inline docs/events/timeline-ui-overview.asciidoc:184]: Changing "alerts" to "events" or "documents" throughout this section as per @paulewing's request via slack.

```suggestion
== Use {esql} to interact with events 
```

File: docs/events/timeline-ui-overview.asciidoc [hunk 6]
```diff
@@ -178,3 +178,29 @@ From the *Correlation* tab, you can also do the following:
...
 * Reorder the columns and choose which fields to display.
 * Choose a data view and whether to show detection alerts only.
+
+[discrete]
+[[esql-in-timeline]]
+== Use {esql} to interact with alerts 
+
+preview::[]
+
+The {ref}/esql.html[Elasticsearch Query Language ({esql})] provides a powerful way to filter, transform, and analyze alert data stored in Elasticsearch. {esql} queries use "pipes" to manipulate and transform data in a step-by-step fashion. This approach allows you to compose a series of operations, where the output of one operation becomes the input for the next, enabling complex data transformations and analysis.
```
Comment (nastasha-solomon) [inline docs/events/timeline-ui-overview.asciidoc:188]:
```suggestion
The {ref}/esql.html[Elasticsearch Query Language ({esql})] provides a powerful way to filter, transform, and analyze event data stored in Elasticsearch. {esql} queries use "pipes" to manipulate and transform data in a step-by-step fashion. This approach allows you to compose a series of operations, where the output of one operation becomes the input for the next, enabling complex data transformations and analysis.
```

File: docs/events/timeline-ui-overview.asciidoc [hunk 7]
```diff
@@ -178,3 +178,29 @@ From the *Correlation* tab, you can also do the following:
...
+
+You can use {esql} in Timeline by opening the **{esql}** tab. From there, you can: 
+
+- Explore your alerts using the default query, or create a custom one. The default query searches the Security alert index (`.alerts-security.alerts-default`) and  indices specified in the <<update-sec-indices,Security data view>> for alerts that were generated during the last 15 minutes. The query's output is limited to the top ten results.
```
Comment (nastasha-solomon) [inline docs/events/timeline-ui-overview.asciidoc:192]:
```suggestion
- Explore your events using the default query, or create a custom one. The default query searches documents within the Security alert index (`.alerts-security.alerts-default`) and indices specified in the <<update-sec-indices,Security data view>>, then returns 10 events from the defined time range.  
```

File: docs/events/timeline-ui-overview.asciidoc [hunk 8]
```diff
@@ -178,3 +178,29 @@ From the *Correlation* tab, you can also do the following:
 * Specify the date and time range that you want to investigate.
 * Reorder the columns and choose which fields to display.
 * Choose a data view and whether to show detection alerts only.
+
+[discrete]
+[[esql-in-timeline]]
+== Use {esql} to interact with events
```
Comment (nastasha-solomon) [inline docs/events/timeline-ui-overview.asciidoc:184]:
```suggestion
== Use {esql} to investigate events 
```

File: docs/events/timeline-ui-overview.asciidoc (hunk 2 above)
Comment (nastasha-solomon) [reply docs/events/timeline-ui-overview.asciidoc]: Resolving this as per our discussion in Slack.

File: docs/events/timeline-ui-overview.asciidoc [hunk 9]
```diff
@@ -178,3 +178,29 @@ From the *Correlation* tab, you can also do the following:
...
+- Visualize query results using {kibana-ref}/discover.html[Discover] functionality.
+
+[role="screenshot"]
+image::images/esql-tab.png[a Timeline's ES|QL tab]
+
+[discrete]
+[[esql-in-timeline-resources]]
+=== Additional {esql} resources 
+
+To get started using {esql}, read the tutorial for {ref}/esql.html[using {esql} in Kibana]. Much of the functionality available in Kibana is also available in Timeline.
```
Comment (abdonpijpelink) [inline docs/events/timeline-ui-overview.asciidoc:203]:
```suggestion
To get started using {esql}, read the tutorial for {ref}/esql-kibana.html[using {esql} in Kibana]. Much of the functionality available in Kibana is also available in Timeline.  
```

File: docs/events/timeline-ui-overview.asciidoc [hunk 10]
```diff
@@ -178,3 +178,29 @@ From the *Correlation* tab, you can also do the following:
...
 * Reorder the columns and choose which fields to display.
 * Choose a data view and whether to show detection alerts only.
+
+[discrete]
+[[esql-in-timeline]]
+== Use {esql} to investigate events 
+
+preview::[]
+
+The {ref}/esql.html[Elasticsearch Query Language ({esql})] provides a powerful way to filter, transform, and analyze event data stored in Elasticsearch. {esql} queries use "pipes" to manipulate and transform data in a step-by-step fashion. This approach allows you to compose a series of operations, where the output of one operation becomes the input for the next, enabling complex data transformations and analysis.
```
Comment (jmikell821) [inline docs/events/timeline-ui-overview.asciidoc:188]:
```suggestion
The {ref}/esql.html[Elasticsearch Query Language ({esql})] provides a powerful way to filter, transform, and analyze event data stored in {es}. {esql} queries use "pipes" to manipulate and transform data in a step-by-step fashion. This approach allows you to compose a series of operations, where the output of one operation becomes the input for the next, enabling complex data transformations and analysis.
```

File: docs/events/timeline-ui-overview.asciidoc [hunk 11]
```diff
@@ -178,3 +178,29 @@ From the *Correlation* tab, you can also do the following:
...
+- Visualize query results using {kibana-ref}/discover.html[Discover] functionality.
+
+[role="screenshot"]
+image::images/esql-tab.png[a Timeline's ES|QL tab]
+
+[discrete]
+[[esql-in-timeline-resources]]
+=== Additional {esql} resources 
+
+To get started using {esql}, read the tutorial for {ref}/esql-kibana.html[using {esql} in Kibana]. Much of the functionality available in Kibana is also available in Timeline.
```
Comment (jmikell821) [inline docs/events/timeline-ui-overview.asciidoc:203]:
```suggestion
To get started using {esql}, read the tutorial for {ref}/esql-kibana.html[using {esql} in {kib}]. Much of the functionality available in {kib} is also available in Timeline.  
```

**Review by jmikell821 — COMMENTED** [review]
Just a few attribute comments; LGTM otherwise 😄

**Pull request merged**